app/routes/events/components/EventAdministrate/index.tsx

```diff
@@ -1,28 +1,36 @@
 import { Content } from 'app/components/Content';
 import { LoginPage } from 'app/components/LoginForm';
 import NavigationTab, { NavigationLink } from 'app/components/NavigationTab';
-import type { EventPool, EventAdministrate } from 'app/models';
+import type { ID } from 'app/store/models';
+import type { AdministrateEvent } from 'app/store/models/Event';
+
+import type { CurrentUser } from 'app/store/models/User';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
 import type { ReactNode } from 'react';
 
 type Props = {
   children: (props: Props) => ReactNode;
-  currentUser: Record<string, any>;
+  currentUser: CurrentUser;
   isMe: boolean;
-  event?: EventAdministrate;
+  event?: AdministrateEvent;
   match: {
     params: {
       eventId: string;
     };
   };
-  pools: Array<EventPool>;
+  pools: Array<ID>;
 };
 
 const EventAdministrateIndex = (props: Props) => {
   const base = `/events/${props.match.params.eventId}/administrate`;
   // At the moment changing settings for other users only works
   // for the settings under `/profile` - so no point in showing
   // the other tabs.
+  const { currentUser, event } = props;
+
+  const canSeeAllergies =
+    currentUser?.id === event?.createdBy?.id ||
+    currentUser?.abakusGroups?.includes(event?.responsibleGroup?.id as ID);
```

Comment:
This should just be on the `actionGrant` instead. You should never do twice the computation (both on server and client).

Reply:
really struggling with putting actionGrant on the response from backend.

Not sure how to deal with it